### ISSUE TITLE
QR Scanner doesn't scan on certain device

### ISSUE THREAD

**Issue by darrenchuah** [googlesamples/android-vision#422]
I've try my QR scanner on my on device and it work without a problem, but when I tried it on another device, everything still runs fine except the QR scanner just won't recognise the QR Code although the camera is running.

My Device: Xiaomi Redmi 3S
Device that couldn't recognise QR Code: Xiaomi Redmi 5

**Comment by grillermo** [comment]
This is probably due to the barcodeDetector.isOperational always returning false issue.
Can you try that on your app, so we can include your case on the bug?

**Comment by darrenchuah** [comment]
> This is probably due to the barcodeDetector.isOperational always returning false issue.
> Can you try that on your app, so we can include your case on the bug?

After clearing out some space, the scanner work like a charm.
Is it possible that android vision have a specific limit on how much storage must be available to be able to work? If so, is there any work around?

**Comment by grillermo** [comment]
Yes there is a 10% limit but there's another bug using the android studio debugger i saw android complaining about google play services version mistmach when trying to download the vision library, i guess deleting the cache resets that sometimes, not always for me.

My always work solution has been to update google play services, then delete the failing app, then restart the device then reinstall the app. It even works when updating to beta versions, get them from here:
https://www.apkmirror.com/?post_type=app_release&searchtype=apk&s=google+play+services

And don't forget to disable google play services automatic updates or it will break your app again, and google complains about android versions fragmentation... 
For some reason manual APK updates do not break it.

**Comment by darrenchuah** [comment]
> Yes there is a 10% limit but there's another bug using the android studio debugger i saw android complaining about google play services version mistmach when trying to download the vision library, i guess deleting the cache resets that sometimes, not always for me.
> 
> My always work solution has been to update google play services, then delete the failing app, then restart the device then reinstall the app. It even works when updating to beta versions, get them from here:
> https://www.apkmirror.com/?post_type=app_release&searchtype=apk&s=google+play+services
> 
> And don't forget to disable google play services automatic updates or it will break your app again, and google complains about android versions fragmentation...
> For some reason manual APK updates do not break it.

Is there a way to bypass the 10% limit?

App user normally doesn't know about the 10% limit, which will kind off a bad rep of not being able to function on their phone.
